tests/software_tests/can/addressing/test_normal_addressing.py

```
@@ -0,0 +1,569 @@
+                   "can_id": mock_encode_can_id.return_value,
+                   "target_address": target_address,
+                   "source_address": source_address,
+                   "address_extension": None,
+               }
+        self.mock_validate_addressing_type.assert_called_once_with(addressing_type)
+        self.mock_validate_raw_byte.assert_has_calls([call(target_address), call(source_address)], any_order=True)
+        mock_encode_can_id.assert_called_once_with(addressing_type=self.mock_validate_addressing_type.return_value,
+                                                   target_address=target_address,
+                                                   source_address=source_address)
+
+    @pytest.mark.parametrize("addressing_type, target_address, source_address, can_id", [
```

Ruff (0.12.2)

Wrong type passed to first argument of `pytest.mark.parametrize`; expected `tuple`

Use a `tuple` for the first argument

(PT006)